services/s3/src/main/resources/codegen-resources/customization.config

```diff
@@ -172,5 +172,13 @@
       "clientConfiguration"
     ]
   },
+  "additionalBuilderMethods": [
+      {
+        "methodName": "crtBuilder",
+        "clientTypes": ["async"],
```

Comment:
minor: should this be singular `clientType` since we know for sure that sync and async don't share the same instanceType and returnType?

Reply:
IMO keeping clientTypes will give us flexibility to add some builders of CustomLibraries which needs to be both in Sync and Async in future.

Reply:
Yeah, but sync and async clients will most likely have different `instanceType`s and `returnType`s, eg: `S3AsyncClientBuilder` and `S3ClientBuilder`, so I think we will probably never use ` "clientTypes": ["async", "sync"]` in reality and we will probably have something like below:


```
"additionalBuilderMethods": [
        {
          "methodName": "builderOne",
          "clientTypes": ["sync"],
          "instanceType": "software.amazon.awssdk.services.builder.DefaultBuilder",
          "returnType": "software.amazon.awssdk.services.builder.CustomBuilder"
        },
        {
          "methodName": "builderOne",
          "clientTypes": ["async"],
          "instanceType": "software.amazon.awssdk.services.builder.DefaultAsyncBuilder",
          "returnType": "software.amazon.awssdk.services.builder.CustomAsyncBuilder"
        }
```

Reply:
Done